### PR TITLE
feat(rpc): net namespace

### PR DESCRIPTION
**Pull request by rkrasiuk**
Create an implementation of `net_` namespace for rpc

### CODE REVIEW

File: crates/net/rpc/src/net/mod.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    fn version(&self) -> Result<String> {
+        Ok(self.network_id.clone())
+    }
+
+    fn peer_count(&self) -> Result<PeerCount> {
+        Ok(PeerCount::Number(self.network.num_connected_peers() as u32))
+    }
+
+    fn is_listening(&self) -> Result<bool> {
+        todo!()
```
Comment (rkrasiuk) [inline crates/net/rpc/src/net/mod.rs:33]: @mattsse @Rjected should this value be returned by the `NetworkHandle`?

File: crates/net/rpc/src/net/mod.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+/// Net rpc implementation
+impl NetApiServer for NetApi {
+    fn version(&self) -> Result<String> {
+        Ok(self.network_id.clone())
+    }
+
+    fn peer_count(&self) -> Result<PeerCount> {
+        Ok(PeerCount::Number(self.network.num_connected_peers() as u32))
```
Comment (mattsse) [inline crates/net/rpc/src/net/mod.rs:29]: iirc the default is hex, perhaps we should make this configurable? and default is as hex?

File: crates/net/rpc/src/net/mod.rs (hunk 1 above)
Comment (mattsse) [reply crates/net/rpc/src/net/mod.rs:33]: I think we can just return true here.
Or perhaps this should be an atomic bool in `NetworkHandle`

File: crates/net/rpc/src/net/mod.rs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+use reth_rpc_types::PeerCount;
+
+/// `Net` API implementation.
+///
+/// This type provides the functionality for handling `net_` related requests.
+pub struct NetApi {
+    /// An interface to interact with the network
+    network: NetworkHandle,
+    /// The devp2p network ID
+    network_id: String,
```
Comment (mattsse) [inline crates/net/rpc/src/net/mod.rs:13]: I suggest we replace this with `EthApi` and use `EthApi::chain_id()` for `fn version`

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/net/rpc/src/net/mod.rs (hunk 3 above)
Comment (rkrasiuk) [reply crates/net/rpc/src/net/mod.rs:13]: dyn EthApi trait of the impl? the latter would require its generics 😓

Comment (mattsse) [reply crates/net/rpc/src/net/mod.rs:13]: Ah right,

okay, so what we should do here is, introduce another `EthApi` trait, but which is a subset of `EthApiServer` trait but without generics, essentially everything that is already in `impl EthApi {`

Comment (mattsse) [reply crates/net/rpc/src/net/mod.rs:13]: Or we could use `Box< dyn EthApiServer>` which is perhaps easier, but I guess we still want a separate `EthApi` trait

Comment (gakonst) [reply crates/net/rpc/src/net/mod.rs:13]: EthApi trait IMO

File: crates/net/rpc/src/net/mod.rs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+use crate::eth::EthApiSpec;
+
+/// `Net` API implementation.
+///
+/// This type provides the functionality for handling `net` related requests.
+pub struct NetApi {
+    /// An interface to interact with the network
+    network: NetworkHandle,
+    /// The implementation of `eth` API
+    eth: Box<dyn EthApiSpec + Send + Sync>,
```
Comment (mattsse) [inline crates/net/rpc/src/net/mod.rs:15]: let's add these bounds to the trait directly.

File: crates/net/rpc/src/net/mod.rs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+use jsonrpsee::core::RpcResult as Result;
+use reth_network::NetworkHandle;
+use reth_rpc_api::NetApiServer;
+use reth_rpc_types::PeerCount;
+
```
Comment (mattsse) [inline crates/net/rpc/src/net/mod.rs:5]:
```suggestion
```

**Review by mattsse — COMMENTED** [review]
smol nit, otherwise lgtm

**Review by mattsse — APPROVED** [review]

**Review by gakonst — APPROVED** [review]

**Pull request merged**